### ISSUE TITLE
Ask questions - suggestions result

### ISSUE THREAD

**Issue by stvlada** [SFWLtd/plg_q2a_integration#1]
Have you been able to reproduce issue where when you try to ask new question, results on questions with existing terms/words show up?
in my case if my external users are true, then I have problem to get results and I'm getting an error not showing these results.

Similar question has been asked year ago, but this was manual integration without using your plugin. Same result with using your plugin.

http://www.question2answer.org/qa/51965/error-displaying-the-error-page-asktitle-php-at-line-76

strange thing is that this works on localhost, but not on actual server...

Thanks

**Comment by Spudley** [comment]
Hi. No, I have not seen this error.

Are the symptoms the same as described in the linked question? From the error message in that question, it sounds like there's a PHP error happening somewhere; you should be able to find out more about the error by checking the server logs. That might give some clues as to the root cause of the issue. Without that, it's not going to be easy to diagnose it.

**Comment by stvlada** [comment]
it is hard to reproduce error log. Error that I'm getting while writing a new question just below the question title field is...
As I said, this is happening only when external users are true.

> Error displaying the error pagea

Those are some errors that I could find

> Call to undefined function qa_get_logged_in_userid() in serverpath/qa-include/ajax/asktitle.php on line 2

> Call to undefined function qa_get_logged_in_userid() in serverpath/qa-include/ajax/asktitle.php on line 3

**Comment by Spudley** [comment]
The error messages you've quoted are strange, because `asktitle.php` has about 20 lines of comments before any actual code, so you shouldn't get an error on line 2 or 3 of that file.

However, if you had edited the file and ripped out top part of it to leave the call to `qa_get_logged_in_userid()` as the first thing in the file, then yes, the error would make sense, because that function is defined in one of the files that is included by the `require_once` statements at the begining of the code, so if you ripped out those `require_once` statements then the `qa_get_logged_in_userid()` function won't be loaded.

In summary, if you're getting the error you've quoted, then it means that you've been editing the core Q2A files and your edits have broken the functionality.

I suggest restoring your Q2A installation to a fresh copy without any core files modified (except as per the instructions in my plugin).

**Comment by stvlada** [comment]
OK, I have tried with fresh copy of joomla and q2a, and I've followed your instructions. I'm not able to reproduce results of the similar questions (suggestions below title field)? Same as described above
Tested on Joomla 3.7.2 and Q2A 1.7.4

few of the latest errors:

> Constant JPATH_BASE already defined in /serverpath/plugins/q2a/qaintegration/qa-external/qa-external-users.php on line 152

> PHP Strict Standards:  Only variables should be passed by reference in /serverpath/plugins/q2a/qaintegration/qa-external/qa-external-users.php on line 167

> PHP Strict Standards:  Only variables should be passed by reference in /serverpath/plugins/q2a/qaintegration/qa-external/qa-external-users.php on line 171

**Comment by Spudley** [comment]
The latter two errors are bugs in Q2A which probably should be fixed. The error itself is because Q2A was originally written for earlier PHP versions, and uses a coding technique that has now been deprecated. It's not a major issue for now but does need to be addressed; I'd suggest raising a bug report with Q2A quoting those error message. But for now you should be able to ignore them by switching error trapping to ignore strict errors. More info on how to do that here: https://stackoverflow.com/questions/9983286/disabling-strict-standards-in-php-5-4.

The first error can also be ignored in the same way by switching off error trapping for notices. This error may be a bug in my plugin or possibly you may be including your own Joomla integration as well as mine? Either way, it is telling you that you've defined JPATH_BASE twice. As long as the definitions are the same, this really doesn't matter and can be ignored.

The other thing I'd suggest is to test it without the plugin. If you can repeat the error without the plugin, then you'll need to raise it with Q2A directly. If it only happens when the plugin is installed, then I'll need to look in more detail. I may not have time to get into it too deeply for a bit though; I've got a lot on at the moment.

**Comment by stvlada** [comment]
I followed link below and now integration works fine, as well as "suggest option" when asking new question. Problem is somewhere around qa-external-users.php in your plugin version. Don't know what to say, but link below resolved my problem for now.

[http://www.question2answer.org/qa/20758/how-to-integrate-joomla-with-q2a-single-sign-on-solved](url)

**Comment by stvlada** [comment]
This has been resolved by updating Q2A.
